keras/utils/test_utils.py

```diff
@@ -42,6 +42,20 @@ def get_test_data(num_train=1000, num_test=500, input_shape=(10,),
     return (X[:num_train], y[:num_train]), (X[num_train:], y[num_train:])
 
 
+def data_generator(x, y, batch_size, multiple_inputs=False):
+    max_batch_index = len(x) // batch_size
+    i = 0
+    while 1:
+        x_batch = x[i * batch_size: (i + 1) * batch_size]
+        y_batch = y[i * batch_size: (i + 1) * batch_size]
+        if multiple_inputs:
+            yield [x_batch, x_batch], [y_batch, y_batch]
```

Comment:
This is way too ad-hoc. The choice of having exactly 2 input and 2 outputs and having the input data / target data be just a repetition of the same array is something that happens to work in some of the existing unit tests, but it wouldn't work in the general case.